hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/StorageDistributionEndpoint.java

```diff
@@ -0,0 +1,199 @@
+  }
+
+  private StorageCapacityDistributionResponse buildStorageDistributionResponse(
+      List<DatanodeStorageReport> nodeStorageReports,
+      GlobalStorageReport storageMetrics,
+      Map<String, Long> namespaceMetrics) {
+    DeletedBlocksTransactionSummary scmSummary;
+    try {
+      scmSummary = scmClient.getDeletedBlockSummary();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    long totalPendingAtDnSide = nodeStorageReports.stream().mapToLong(DatanodeStorageReport::getPendingDeletions).sum();
+
+    DeletionPendingBytesByStage deletionPendingBytesByStage =
+        createDeletionPendingBytesByStage(namespaceMetrics.get("totalPendingAtOmSide"),
+            scmSummary.getTotalBlockReplicatedSize(),
+            totalPendingAtDnSide);
+    return StorageCapacityDistributionResponse.newBuilder()
+        .setDataNodeUsage(nodeStorageReports)
+        .setGlobalStorage(storageMetrics)
+        .setGlobalNamespace(new GlobalNamespaceReport(namespaceMetrics.get("totalUsedNamespace"), 0))
+        .setUsedSpaceBreakDown(new UsedSpaceBreakDown(
+            namespaceMetrics.get("totalOpenKeySize"),
+            namespaceMetrics.get("totalCommittedSize"),
+            deletionPendingBytesByStage))
+        .build();
+  }
+
+  private List<DatanodeStorageReport> collectDatanodeReports() {
+    return nodeManager.getAllNodes().stream()
+        .map(this::getStorageReport)
+        .collect(Collectors.toList());
+  }
+
+  private long calculatePendingSizes() {
+    Map<String, Long> pendingDeletedDirSizes = new HashMap<>();
+    omdbInsightEndpoint.calculateTotalPendingDeletedDirSizes(pendingDeletedDirSizes);
+    Map<String, Long> pendingKeySize = new HashMap<>();
+    omdbInsightEndpoint.createKeysSummaryForDeletedKey(pendingKeySize);
+    long totalPendingDeletedDirSize = pendingDeletedDirSizes.getOrDefault("totalReplicatedDataSize", 0L);
+    long totalPendingKeySize = pendingKeySize.getOrDefault("totalReplicatedDataSize", 0L);
+    return totalPendingDeletedDirSize + totalPendingKeySize;
+  }
+
+  private long calculateOpenKeySizes() {
+    Map<String, Long> openKeySummary = new HashMap<>();
+    omdbInsightEndpoint.createKeysSummaryForOpenKey(openKeySummary);
+    omdbInsightEndpoint.createKeysSummaryForOpenMPUKey(openKeySummary);
+    Map<String, Long> pendingKeySize = new HashMap<>();
+    omdbInsightEndpoint.createKeysSummaryForDeletedKey(pendingKeySize);
+    long openKeyDataSize = openKeySummary.getOrDefault("totalReplicatedDataSize", 0L);
+    long totalMPUKeySize = pendingKeySize.getOrDefault("totalDataSize", 0L);
+    return openKeyDataSize + totalMPUKeySize;
+  }
+
+  private long calculateCommittedSize() {
+    try {
+      Response rootResponse = nsSummaryEndpoint.getDiskUsage("/", false, true, false);
+      DUResponse duRootRes = (DUResponse) rootResponse.getEntity();
+      return duRootRes.getSizeWithReplica();
+    } catch (IOException e) {
+      return -1;
+    }
+  }
```

Comment:
Returns magic number -1 on error without logging
No validation of response status or entity

can replace with - 

```
private long calculateCommittedSize() {
  try {
    Response rootResponse = nsSummaryEndpoint.getDiskUsage("/", false, true, false);
    if (rootResponse.getStatus() != Response.Status.OK.getStatusCode()) {
      LOG.warn("Failed to get disk usage, status: {}", rootResponse.getStatus());
      return 0L;
    }
    DUResponse duRootRes = (DUResponse) rootResponse.getEntity();
    return duRootRes != null ? duRootRes.getSizeWithReplica() : 0L;
  } catch (IOException e) {
    LOG.error("IOException while calculating committed size", e);
    return 0L;
  }
}
```